### DISCUSSION TITLE
Random SSL Error

### DISCUSSION THREAD

**Discussion by devchris2303** [postalserver/postal#2983] [Help with using Postal]
The vast majority of emails coming into our postal instance are absolutely fine, roughly around 2-3 thousand inbound emails a day.

We are getting the attached error.

One instance of the error that is interesting, is that, the sender sent this email to 2 email addresses in the CC that are defined within postal...it was received by one domain and sent through the route, but the other one was never received and the attached error was thrown. I can see the email being received in the next log entries.

[errorOutput.txt](https://github.com/postalserver/postal/files/15402484/errorOutput.txt)

Any advice welcome!

**Comment by willpower232** [comment]
> Errno::ECONNRESET: Connection reset by peer - SSL_accept

I'm sure I see this from time to time, I've presumed it is one end or the other struggling to deal with the length of the connection happening.

Can you elaborate more on what is sending this message? Did it retry on the second send?

**Reply by devchris2303** [reply]
This is a regular email from a human sent from Outlook, through Mimecast and then into postal. There was no retry as it wasn't even received into postal, however the other email went in fine. 

Just to elaborate; The email is from user@mydomain.com sent to (in the CC)  ID@systemone.com; ID@systemtwo.com

Both systemone.com and systemtwo.com domains are set up in Postal and receive mail perfectly fine, however the systemone.com email was never received into postal [and threw that attached error] the systemtwo.com email was received perfectly fine and was imported into its respective system with no issue

**Reply by devchris2303** [reply]
[moreError.txt](https://github.com/postalserver/postal/files/15402764/moreError.txt)

Just some further logs for you - there isn't even a second between the connection opening and it appearing to be dropped...however the next connection from the same source connects fine.

**Reply by devchris2303** [reply]
Just a further note, 

I re-added the cert for TLS and rebooted, I'm not seeing any instance of the error anymore. Perhaps it was that... however it wouldn't explain by one domain received it but not the other from the same sender? I'm not sure. 

I'll keep an eye on it and report back.
